### ISSUE TITLE
recursive put fails with 0.5.5

### ISSUE THREAD

**Issue by mgsnuno** [dask/adlfs#123]
```python
abfs_storage_options = {
    "account_name": os.environ["pipelines_abfs_name"],
    "account_key": os.environ["pipelines_abfs_key"],
}
fs = AzureBlobFileSystem(**abfs_storage_options)
fs.put('/tmp/foldertocopy/', 'container/folder/', recursive=True)  # fails with Invalid '/tmp/foldertocopy/' is a directory
```
**What happened**:
With latest adlfs 0.5.5 copying a local folder recursively fails. With 0.3.3 no problem (0.3.3 was before asyncio).

**Environment**:

- adlfs version: 0.5.5
- Dask version: 2.30.0
- Python version: 3.8.6
- Operating System: linux
- Install method (conda, pip, source): conda

**Comment by mgsnuno** [comment]
@hayesgb did you manage to replicate the issue? Thanks

**Comment by hayesgb** [comment]
@mgsnuno -- I'll pull #126 in later today.  Appreciate any feedback.

**Comment by mgsnuno** [comment]
@hayesgb I tested like I mentioned above, actually on a folder that was causing the `invalid empty header` error, and is working fine with master/c95c0512fda523bf79e04bf2c667e6df40f0351d

**Comment by mgsnuno** [comment]
fixed in v0.5.6

**Comment by mgsnuno** [comment]
fixed in v0.5.7
